sdks/python/apache_beam/io/textio.py

```diff
@@ -78,6 +78,10 @@ def position(self, value):
                          'size of data %d.', value, len(self._data))
       self._position = value
 
+    def reset(self):
+      self.data = ''
```

Comment:
<!--new_thread; commit:70ade9d4fe6a8e31ff0be4d39b731510c93b4bf9; resolved:0-->
I don't think it makes sense to add a reset() method since we are only creating this object at max 3 times. It should be fine to just create new objects and replace the original.

Reply:
The reason for this is that we want to be able to use the same buffer in the _read_records and _process_header methods. Setting read_buffer = ReadBuffer('', 0) won't behave the same in this case since it will only reset the buffer in the scope of the _process_header method but not reset the buffer in the caller of _process_header.